addons/JobChange/jobchange.lua

```diff
@@ -0,0 +1,218 @@
+            ["Main Job"]=0,
+            ["Sub Job"]=job,
+            ["_unknown1"]=0,
+            ["_unknown2"]=0
+            })
+            packets.inject(packet)
+            coroutine.sleep(0.5)
+        end
+    end
+end
+
+windower.register_event('addon command', function(command, ...)
+    local args = L{...}
+    local job = ''
+    if args[1] then 
+        job = args[1]:lower()
+    end
+    local currentjob = windower.ffxi.get_player()
+    local main_sub = ''
+    if command:lower() == 'main' then
+        main_sub = 'main'
+    elseif command:lower() == 'sub' then
+        main_sub = 'sub'
+    elseif command:lower() == 'reset' then
+        windower.add_to_chat(4,"JobChange: Resetting Job")
+        main_sub = 'sub'
+        job = windower.ffxi.get_player().sub_job:lower()
+    else
+        windower.add_to_chat(4,"JobChange Syntax: //jc main|sub JOB  -- Chnages main or sub to target JOB")
+        windower.add_to_chat(4,"JobChange Syntax: //jc reset -- Resets Current Job")
+        return
+    end
+    local conflict = find_conflict(job)
+    local jobid = find_job(job)
+    if jobid then 
+        local npc = find_job_change_npc()
+        if npc then
+            if not conflict then 
+                jobchange(jobid,main_sub)
+            else
+                local temp_job = find_temp_job()            
+                windower.add_to_chat(4,"JobChange: Conflict with "..conflict)
+                if main_sub == conflict then 
+                    jobchange(temp_job,main_sub)
+                    jobchange(jobid,main_sub)
+                else
+                    jobchange(temp_job,conflict)
+                    jobchange(jobid,main_sub)
+                end
+            end
+        else
+            windower.add_to_chat(4,"JobChange: Not close enough to a Moogle!")
+        end        
+    else
+        windower.add_to_chat(4,"JobChange: Could not change "..command.." to "..job:upper().." ---Mistype|NotUnlocked")
+    end
+    
+end)
+
+function find_conflict(job)
+    local self = windower.ffxi.get_player()
+    if self.main_job == job:upper() then
+        return "main"
+    end
+    if self.sub_job == job:upper() then
+        return "sub"
+    end
+end
+
+function find_temp_job()
+    local starting_jobs = {
+    -- WAR, MNK, WHM, BLM, THF, RDM - main starting jobs.
+        ["WAR"] = 1,
+        ["MNK"] = 2,
+        ["WHM"] = 3,
+        ["BLM"] = 4,
+        ["RDM"] = 5,
+        ["THF"] = 6,
+    }
+    for index,value in pairs(starting_jobs) do
+        if not find_conflict(index) then 
+            return value
+        end
+    end
+end
+
+function find_job(job)
+    local self = windower.ffxi.get_player()
+    local ki = windower.ffxi.get_key_items() 
+    
+    for index,value in pairs(res.jobs) do
+        if value.ens:lower() == job then 
+            jobid = index
+        end
+    end
+    --[[
+        Small rant/request/can't figure out a better way.
```

Comment:
There was no better way. But there is now! I just adjusted the new hook to gather that info and *LuaCore* now reports level `0` for jobs that are not unlocked yet.